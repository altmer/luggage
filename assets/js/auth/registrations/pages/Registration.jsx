/* eslint-disable react/no-unknown-property */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Translate } from 'react-i18nify';

import yup from 'yup';
import Form from 'react-formal';

import { signUp } from '../ducks';
import errorMessages from '../../../services/errorMessages';
import LocalizedFormMessage from '../../../components/LocalizedFormMessage';

export class Registration extends React.Component {
  constructor(props) {
    super(props);
    this.submit = this.submit.bind(this);
    this.userSchema = yup.object({
      name: yup.string().default('').required('required'),
      email: yup.string().default('').required('required').matches(/@/, 'email'),
      password: yup.string().default('').required('required').min(5, 'passwordLength'),
      password_confirmation: yup.string().default('').required('required'),
    });
  }

  submit(formValue) {
    this.props.signUp(formValue);
  }

  render() {
    const errors = errorMessages(this.props.errors);

    return (
      <div className="container">
        <div className="row justify-content-center">
          <div className="col-md-12 col-lg-8 col-xl-6">
            <Form
              schema={this.userSchema}
              defaultValue={this.userSchema.default()}
              onSubmit={this.submit}
            >
              <div className="form-group">
                <label htmlFor="first_name">
                  <Translate value="registrations.new.name" />
                </label>
                <Form.Field name="name" className="form-control" />
                <LocalizedFormMessage for="name" className="error-message" />
                {errors.name}
              </div>
              <div className="form-group">
                <label htmlFor="email">
                  <Translate value="registrations.new.email" />
                </label>
                <Form.Field name="email" type="email" className="form-control" />
                <LocalizedFormMessage for="email" className="error-message" />
                {errors.email}
              </div>
              <div className="form-group">
                <label htmlFor="password">
                  <Translate value="registrations.new.password" />
                </label>
                <Form.Field name="password" type="password" className="form-control" />
                <LocalizedFormMessage for="password" className="error-message" />
                {errors.password}
              </div>
              <div className="form-group">
                <label htmlFor="password_confirmation">
                  <Translate value="registrations.new.passwordConfirmation" />
                </label>
                <Form.Field name="password_confirmation" type="password" className="form-control" />
                <LocalizedFormMessage for="password_confirmation" className="error-message" />
                {errors.password_confirmation}
              </div>
              <div className="form-group ">
                <Form.Button type="submit" className="btn btn-outline-success btn-lg btn-block">
                  <Translate value="registrations.new.save" />
                </Form.Button>
              </div>
              <div className="form-group ">
                <a href="/auth/google" className="btn btn-primary btn-google btn-lg btn-block">
                  <Translate value="registrations.new.google" />
                </a>
              </div>
            </Form>
          </div>
        </div>
      </div>
    );
  }
}

Registration.propTypes = {
  errors: PropTypes.shape({
    first_name: PropTypes.arrayOf(PropTypes.string),
    last_name: PropTypes.arrayOf(PropTypes.string),
    email: PropTypes.arrayOf(PropTypes.string),
    password: PropTypes.arrayOf(PropTypes.string),
    password_confirmation: PropTypes.arrayOf(PropTypes.string),
  }).isRequired,
  signUp: PropTypes.func.isRequired,
};

const mapStateToProps = state => (
  {
    errors: state.registrations.errors,
  }
);

const mapDispatchToProps = dispatch => (
  {
    signUp(data) {
      dispatch(signUp(data));
    },
  }
);

export default connect(mapStateToProps, mapDispatchToProps)(Registration);
