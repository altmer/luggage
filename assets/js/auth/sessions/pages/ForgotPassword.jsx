/* eslint-disable react/no-unknown-property */
import React from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';
import { Translate } from 'react-i18nify';

import yup from 'yup';
import Form from 'react-formal';

import { resetPassword } from '../ducks';
import LocalizedFormMessage from '../../../components/LocalizedFormMessage';

export class ForgotPassword extends React.Component {
  constructor(props) {
    super(props);
    this.submit = this.submit.bind(this);
    this.schema = yup.object({
      email: yup.string().default('').required('required'),
    });
  }

  submit(formValue) {
    this.props.resetPassword(formValue.email);
  }

  render() {
    return (
      <div className="container">
        <div className="row justify-content-center">
          <div className="col-xs-12 col-sm-8 col-md-6">
            <Form schema={this.schema} onSubmit={this.submit} className="login-form">
              <div className="form-group">
                <label htmlFor="email">
                  <Translate value="registrations.new.email" />
                </label>
                <Form.Field name="email" type="email" className="form-control" />
                <LocalizedFormMessage for="email" className="error-message" />
              </div>
              <div className="form-group">
                <Form.Button type="submit" className="btn btn-outline-success btn-lg btn-block">
                  <Translate value="sessions.forgot.action" />
                </Form.Button>
              </div>
            </Form>
          </div>
        </div>
      </div>
    );
  }
}

ForgotPassword.propTypes = {
  resetPassword: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => (
  {
    resetPassword(email) {
      dispatch(resetPassword(email));
    },
  }
);

export default connect(() => ({}), mapDispatchToProps)(ForgotPassword);
