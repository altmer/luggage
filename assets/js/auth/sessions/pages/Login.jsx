/* eslint-disable react/no-unknown-property */
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { Translate } from 'react-i18nify';

import yup from 'yup';
import Form from 'react-formal';

import { signIn } from '../ducks';

export class Login extends React.Component {
  constructor(props) {
    super(props);
    this.submit = this.submit.bind(this);
    this.loginSchema = yup.object({
      email: yup.string(),
      password: yup.string(),
    });
  }

  submit(formValue) {
    this.props.signIn(formValue.email, formValue.password);
  }

  render() {
    const { errors } = this.props;

    return (
      <div className="container">
        <div className="row justify-content-center">
          <div className="col-md-12 col-lg-8 col-xl-6">
            <Form schema={this.loginSchema} onSubmit={this.submit} className="login-form">
              <div className="form-group">
                <a href="/auth/google" className="btn btn-primary btn-google btn-lg btn-block">
                  <Translate value="sessions.new.google" />
                </a>
              </div>
              <div className="form-group">
                <label htmlFor="email">
                  <Translate value="registrations.new.email" />
                </label>
                <Form.Field name="email" type="email" className="form-control" />
                <span className="error-message">
                  {errors.error}
                </span>
              </div>
              <div className="form-group">
                <label htmlFor="password">
                  <Translate value="registrations.new.password" />
                </label>
                <Form.Field name="password" type="password" className="form-control" />
              </div>
              <p>
                <Link to="/password/forgotten">
                  <Translate value="sessions.new.forgot" />
                </Link>
              </p>
              <div className="form-group">
                <Form.Button type="submit" className="btn btn-outline-success btn-lg btn-block">
                  <Translate value="sessions.new.save" />
                </Form.Button>
              </div>
            </Form>
          </div>
        </div>
      </div>
    );
  }
}

Login.propTypes = {
  // TODO: refactor this
  errors: PropTypes.shape({
    error: PropTypes.string,
  }).isRequired,
  signIn: PropTypes.func.isRequired,
};

const mapStateToProps = state => (
  { errors: state.session.errors }
);

const mapDispatchToProps = dispatch => (
  {
    signIn(email, password) {
      dispatch(signIn(email, password));
    },
  }
);

export default connect(mapStateToProps, mapDispatchToProps)(Login);
