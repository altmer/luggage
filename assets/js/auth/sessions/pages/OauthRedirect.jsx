import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { signInOauth, signInOauthFailed } from '../ducks';

const OauthRedirect = (props) => {
  const params = new URLSearchParams(location.search);

  if (params.get('result') === 'failure') {
    props.signInOauthFailed();
  } else if (params.get('result') === 'success') {
    props.signInOauth(params.get('token'));
  } else {
    // wtf?
    props.signInOauthFailed();
  }

  return <div />;
};

OauthRedirect.propTypes = {
  signInOauth: PropTypes.func.isRequired,
  signInOauthFailed: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => (
  {
    signInOauth(token) {
      dispatch(signInOauth(token));
    },
    signInOauthFailed() {
      dispatch(signInOauthFailed());
    },
  }
);

export default connect(() => ({}), mapDispatchToProps)(OauthRedirect);
