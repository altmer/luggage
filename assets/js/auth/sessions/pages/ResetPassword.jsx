/* eslint-disable react/no-unknown-property */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Translate } from 'react-i18nify';

import yup from 'yup';
import Form from 'react-formal';

import errorMessages from '../../../services/errorMessages';
import { clearErrors, updatePassword } from '../ducks';
import LocalizedFormMessage from '../../../components/LocalizedFormMessage';

export class ResetPassword extends React.Component {
  constructor(props) {
    super(props);
    this.submit = this.submit.bind(this);
    this.schema = yup.object({
      password: yup.string().default('').required('required').min(5, 'passwordLength'),
      password_confirmation: yup.string().default('').required('required'),
    });
  }

  componentDidMount() {
    this.props.clearErrors();
  }

  submit(formValue) {
    this.props.updatePassword(this.props.token, formValue);
  }

  render() {
    const errors = errorMessages(this.props.errors);

    return (
      <div className="container">
        <div className="row justify-content-center">
          <div className="col-xs-12 col-sm-8 col-md-6">
            <Form
              schema={this.schema}
              defaultValue={this.schema.default()}
              onSubmit={this.submit}
            >
              <div className="form-group">
                <label htmlFor="password">
                  <Translate value="registrations.new.password" />
                </label>
                <Form.Field name="password" type="password" className="form-control" />
                <LocalizedFormMessage for="password" className="error-message" />
                {errors.password}
              </div>
              <div className="form-group">
                <label htmlFor="password_confirmation">
                  <Translate value="registrations.new.passwordConfirmation" />
                </label>
                <Form.Field name="password_confirmation" type="password" className="form-control" />
                <LocalizedFormMessage for="password_confirmation" className="error-message" />
                {errors.password_confirmation}
              </div>
              <div className="form-group">
                <Form.Button type="submit" className="btn btn-outline-success btn-lg btn-block">
                  <Translate value="users.changePasswordForm.save" />
                </Form.Button>
              </div>
            </Form>
          </div>
        </div>
      </div>
    );
  }
}

ResetPassword.propTypes = {
  errors: PropTypes.shape({
    password: PropTypes.arrayOf(PropTypes.string),
    password_confirmation: PropTypes.arrayOf(PropTypes.string),
  }).isRequired,
  updatePassword: PropTypes.func.isRequired,
  clearErrors: PropTypes.func.isRequired,
  token: PropTypes.string.isRequired,
};

const mapStateToProps = (state, ownProps) => (
  {
    token: ownProps.match.params.token,
    errors: state.session.errors,
  }
);

const mapDispatchToProps = dispatch => (
  {
    updatePassword(token, data) {
      dispatch(updatePassword(token, data));
    },
    clearErrors() {
      dispatch(clearErrors());
    },
  }
);

export default connect(mapStateToProps, mapDispatchToProps)(ResetPassword);
