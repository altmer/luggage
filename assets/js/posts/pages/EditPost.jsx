import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';
import { push } from 'react-router-redux';

import PageLoading from '../../components/PageLoading';
import { fetchPost, clearErrors, updatePost } from '../editDucks';
import PostForm from '../components/PostForm';

export class EditPost extends Component {
  // TODO: consider using @autobind
  constructor(props) {
    super(props);
    this.updatePost = this.updatePost.bind(this);
    this.cancel = this.cancel.bind(this);
  }

  componentWillMount() {
    this.props.clearErrors();
  }

  componentDidMount() {
    this.props.fetchPost(this.props.postId);
  }

  updatePost(data) {
    this.props.updatePost(this.props.postId, data);
  }

  cancel() {
    this.props.cancel(this.props.postId);
  }

  render() {
    if (this.props.postFetching) {
      return (<PageLoading />);
    }
    return (
      <PostForm
        onSave={this.updatePost}
        onCancel={this.cancel}
        errors={this.props.errors}
        {...this.props.post}
      />
    );
  }
}

EditPost.propTypes = {
  postId: PropTypes.string.isRequired,
  postFetching: PropTypes.bool.isRequired,
  post: PropTypes.shape({}),
  errors: PropTypes.shape({}),

  fetchPost: PropTypes.func.isRequired,
  updatePost: PropTypes.func.isRequired,
  clearErrors: PropTypes.func.isRequired,
  cancel: PropTypes.func.isRequired,
};

EditPost.defaultProps = {
  post: undefined,
  errors: undefined,
};

const mapStateToProps = (state, ownProps) => (
  {
    postId: ownProps.match.params.postId,
    postFetching: state.postsShow.postFetching,
    post: state.postsShow.currentPost,
    errors: state.postsShow.formErrors,
  }
);

const mapDispatchToProps = dispatch => (
  {
    fetchPost: postId => dispatch(fetchPost(postId)),
    clearErrors: () => dispatch(clearErrors()),
    updatePost: (postId, data) => dispatch(updatePost(postId, data)),
    cancel(postId) {
      dispatch(push(`/posts/${postId}/show`));
    },
  }
);

export default connect(mapStateToProps, mapDispatchToProps)(EditPost);
