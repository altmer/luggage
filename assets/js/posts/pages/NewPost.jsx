import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';
import { push } from 'react-router-redux';

import PostForm from '../components/PostForm';
import { createPost, clearErrors } from '../editDucks';

export class NewPost extends Component {
  componentWillMount() {
    this.props.clearErrors();
  }

  render() {
    return (
      <PostForm
        onSave={this.props.createPost}
        onCancel={this.props.cancel}
        errors={this.props.errors}
      />
    );
  }
}

NewPost.propTypes = {
  errors: PropTypes.shape({}).isRequired,
  createPost: PropTypes.func.isRequired,
  clearErrors: PropTypes.func.isRequired,
  cancel: PropTypes.func.isRequired,
};

const mapStateToProps = state => (
  {
    errors: state.postsShow.formErrors,
  }
);

const mapDispatchToProps = dispatch => (
  {
    createPost(data) {
      dispatch(createPost(data));
    },
    clearErrors() {
      dispatch(clearErrors());
    },
    cancel() {
      dispatch(push('/'));
    },
  }
);

export default connect(mapStateToProps, mapDispatchToProps)(NewPost);
