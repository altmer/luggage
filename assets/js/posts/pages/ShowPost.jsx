import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Translate, I18n } from 'react-i18nify';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import PageLoading from '../../components/PageLoading';
import AddComment from '../components/AddComment';
import Comments from '../components/Comments';
import PostDate from '../components/PostDate';
import User from '../../users/components/User';
import LoadMore from '../../components/LoadMore';
import { confirmAction } from '../../services/confirmation';
import { fetchPost, deletePost, fetchComments, fetchMoreComments, commentReceived } from '../editDucks';

export class ShowPost extends Component {
  constructor(props) {
    super(props);
    this.deletePost = this.deletePost.bind(this);
    this.addComment = this.addComment.bind(this);
    this.canBeEdited = this.canBeEdited.bind(this);
    this.joinChannel = this.joinChannel.bind(this);
    this.fetchMoreComments = this.fetchMoreComments.bind(this);
    this.fetchAll = this.fetchAll.bind(this);

    this.state = {
      channel: null,
    };
  }

  componentDidMount() {
    this.fetchAll();
  }

  componentWillReceiveProps() {
    this.joinChannel();
  }

  componentDidUpdate(nextProps) {
    if (nextProps.postId !== this.props.postId) {
      this.fetchAll();
      this.joinChannel();
    }
  }

  componentWillUnmount() {
    if (this.state.channel) {
      this.state.channel.leave();
    }
  }

  fetchAll() {
    this.props.fetchPost(this.props.postId, this.props.socket);
    this.props.fetchComments(this.props.postId);
  }

  joinChannel() {
    if (!this.props.socket) {
      return;
    }
    if (this.state.channel) {
      this.state.channel.leave();
    }
    const channel = this.props.socket.channel(`post:${this.props.postId}`);
    channel.join();
    channel.on('comment:created', (msg) => {
      this.props.commentReceived(msg.comment);
    });
    this.setState({ channel });
  }

  canBeEdited() {
    if (!this.props.post.user || !this.props.currentUser) {
      return false;
    }
    return this.props.post.user.id === this.props.currentUser.id;
  }

  fetchMoreComments() {
    this.props.fetchMoreComments(this.props.page + 1, this.props.postId);
  }

  deletePost() {
    confirmAction(this.props.deletePost.bind(this, this.props.postId),
                  I18n.t('posts.deleteConfirmation'));
  }

  addComment(text) {
    if (!this.state.channel) {
      return;
    }
    // not enough graceful error and timeout handling
    this.state.channel.push('comment:add', { text }, 10000);
  }

  render() {
    if (this.props.postFetching || !this.props.post.title) {
      return (<PageLoading />);
    }
    return (
      <div className="container post-page">
        <div className="row">
          <div className="col-12">
            <div className="post-show">
              <div className="post-author">
                <User {...this.props.post.user}>
                  <PostDate locale={this.props.locale} date={this.props.post.inserted_at} />
                </User>
              </div>

              <h1 className="post-title display-4">{this.props.post.title}</h1>
              {this.canBeEdited() &&
                <span className="post-actions">
                  <Link to={`/posts/${this.props.postId}/edit`}>
                    <Translate value="posts.edit" />
                  </Link>
                  <a href="#0" onClick={this.deletePost} className="btn-delete-post">
                    <Translate value="posts.delete" />
                  </a>
                </span>
              }
              <p className="post-body">{this.props.post.body}</p>
            </div>
          </div>
        </div>
        <div className="comments-show">
          <h4 className="comments-show-title">
            <Translate value="posts.comments.title" />
          </h4>
          {this.props.currentUser && this.state.channel &&
            <AddComment onSave={this.addComment} />
          }
          <Comments
            comments={this.props.comments}
            locale={this.props.locale}
            fetching={this.props.commentsFetching}
          />
          <LoadMore
            show={this.props.hasMore}
            fetching={this.props.fetchingMore}
            onClick={this.fetchMoreComments}
          />
        </div>
      </div>
    );
  }
}

ShowPost.propTypes = {
  socket: PropTypes.shape({
    channel: PropTypes.func,
  }),

  post: PropTypes.shape({
    title: PropTypes.string,
    body: PropTypes.string,
    inserted_at: PropTypes.string,
    user: PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
    }),
  }),

  postId: PropTypes.string.isRequired,
  postFetching: PropTypes.bool.isRequired,
  currentUser: PropTypes.shape({
    id: PropTypes.number,
  }),
  locale: PropTypes.string.isRequired,

  deletePost: PropTypes.func.isRequired,
  fetchPost: PropTypes.func.isRequired,

  fetchComments: PropTypes.func.isRequired,
  comments: PropTypes.arrayOf(PropTypes.shape({})),
  commentsFetching: PropTypes.bool.isRequired,
  commentReceived: PropTypes.func.isRequired,
  fetchMoreComments: PropTypes.func.isRequired,
  page: PropTypes.number,
  hasMore: PropTypes.bool,
  fetchingMore: PropTypes.bool,
};

ShowPost.defaultProps = {
  currentUser: undefined,
  post: {},
  comments: [],
  socket: null,
  page: 1,
  hasMore: false,
  fetchingMore: false,
};

const mapStateToProps = (state, ownProps) => (
  {
    socket: state.session.socket,
    postId: ownProps.match.params.postId,
    postFetching: state.postsShow.postFetching,
    post: state.postsShow.currentPost,
    currentUser: state.session.currentUser,
    locale: state.session.locale,
    comments: state.postsShow.comments,
    commentsFetching: state.postsShow.commentsFetching,
    page: state.postsShow.commentsPage,
    hasMore: state.postsShow.hasMoreComments,
    fetchingMore: state.postsShow.commentsFetchingMore,
  }
);

export default connect(
  mapStateToProps,
  { fetchPost, deletePost, fetchComments, commentReceived, fetchMoreComments })(ShowPost);
