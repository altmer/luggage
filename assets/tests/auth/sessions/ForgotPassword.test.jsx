import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { ForgotPassword } from '../../../js/auth/sessions/pages/ForgotPassword';

test('ForgotPassword renders form with email field', () => {
  const subject = renderer.create(
    <ForgotPassword
      resetPassword={jest.fn()}
    />);
  expect(subject.toJSON()).toMatchSnapshot();
});

test('ForgotPassword form submits passwords and token', () => {
  const resetPasswordMock = jest.fn();
  const subject = shallow(
    <ForgotPassword
      resetPassword={resetPasswordMock}
    />);
  subject.instance().submit({
    email: 'email@mail.test',
  });
  expect(resetPasswordMock).toHaveBeenCalledWith('email@mail.test');
});
