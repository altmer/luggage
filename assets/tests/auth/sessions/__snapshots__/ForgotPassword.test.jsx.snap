// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword renders form with email field 1`] = `
<div
  className="container"
>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-xs-12 col-sm-8 col-md-6"
    >
      <form
        className="login-form"
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="email"
          >
            <span
              className={undefined}
              style={undefined}
            >
              Email
            </span>
          </label>
          <input
            className="form-control"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            type="email"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <button
            className="btn btn-outline-success btn-lg btn-block"
            onClick={[Function]}
          >
            <span
              className={undefined}
              style={undefined}
            >
              Action
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
